viewing_party/party.py

```diff
@@ -1,23 +1,185 @@
 # ------------- WAVE 1 --------------------
-
+#---Wave_1_function_1---
 def create_movie(title, genre, rating):
-    pass
 
-# -----------------------------------------
+    movie_dict = {}
+
+    if not title:
+        return None
+    elif not genre:
+        return None
+    elif not rating:
+        return None
+    else: 
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+
+    return movie_dict
+
+create_movie("Title", "Genre", "Rating")
+
+
+# #---Wave_1_function_2---
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+
+add_to_watched({"watched": []}, {
+        "title": "MOVIE_TITLE_1",
+        "genre": "GENRE_1",
+        "rating": "RATING_1"})
+
+
+# #---Wave_1_function_3---
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+
+#---Wave_1_function_4---
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title != (movie["title"]):
+            continue
+        else:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+            print (user_data)
+            return user_data
+    return user_data
+
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+# ---Wave_2_function_1---
 
+def get_watched_avg_rating(user_data):
+
+    ratings = []
+    
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    
+    for movie in user_data["watched"]:
+        ratings.append((movie["rating"]))
+    
+    ratings_sum = sum(ratings)
+    ratings_average = (ratings_sum) / len(ratings)
+
+    return ratings_average
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
 
+# ---Wave_2_function_2---
+def get_most_watched_genre(user_data):
+
+    movie_genre_dict = {}
+
+    if len(user_data["watched"]) == 0:
+        return None
+    
+    for movie in user_data["watched"]:
+
+        
+        title = movie["title"]
+        genre = movie["genre"]
+        
+        if genre not in movie_genre_dict:
+            movie_genre_dict[genre] = []
         
-# -----------------------------------------
+        movie_genre_dict[genre].append(title)
+
+    max_genre = None
+    max_count = 0
+
+    for genre in movie_genre_dict:
+        genre_length = len(movie_genre_dict[genre])
+        if genre_length >= max_count:
+            max_genre = genre
+            max_count = len(movie_genre_dict.values())

```

Comment:
Hmm... Looking through your logic, I see that if a `genre`'s length is bigger than `max_count`, then this `genre` is the new `max_genre` and its length is the new `max_count`. I would expect this line to be:

```python
            max_count = len(movie_genre_dict[genre])
```

This code also passes the tests!